code/core/write_sdl_ntuple.cc

```diff
@@ -6070,20 +6057,14 @@ void printMDs(SDL::Event* event)
     SDL::miniDoublets& miniDoubletsInGPU = (*event->getMiniDoublets());
     SDL::hits& hitsInGPU = (*event->getHits());
     SDL::modules& modulesInGPU = (*event->getModules());
+    SDL::objectRanges& rangesInGPU = (*event->getRanges());
+
+    // Then obtain the lower module index
     for (unsigned int idx = 0; idx <= *(modulesInGPU.nLowerModules); ++idx)
     {
-        for (unsigned int jdx = 0; jdx < miniDoubletsInGPU.nMDs[2*idx]; jdx++)
-        {
-            unsigned int mdIdx = (2*idx) * N_MAX_MD_PER_MODULES + jdx;
-            unsigned int LowerHitIndex = miniDoubletsInGPU.anchorHitIndices[mdIdx];
-            unsigned int UpperHitIndex = miniDoubletsInGPU.outerHitIndices[mdIdx];
-            unsigned int hit0 = hitsInGPU.idxs[LowerHitIndex];
-            unsigned int hit1 = hitsInGPU.idxs[UpperHitIndex];
-            std::cout <<  "VALIDATION 'MD': " << "MD" <<  " hit0: " << hit0 <<  " hit1: " << hit1 <<  std::endl;
-        }
-        for (unsigned int jdx = 0; jdx < miniDoubletsInGPU.nMDs[2*idx+1]; jdx++)
+        for (unsigned int iMD = 0; iMD < miniDoubletsInGPU.nMDs[idx]; iMD++)
         {
-            unsigned int mdIdx = (2*idx+1) * N_MAX_MD_PER_MODULES + jdx;
+            unsigned int mdIdx = rangesInGPU.miniDoubletModuleIndices[idx] + iMD;
```

Comment:
Does this change mean that we were getting the MDs wrong in the output NTuple until now?

Reply:
I'm not sure about do we get it correctly before... This is changing in a function called "printMDs", but we never use this function for a long time actually, I'm not sure even it's still callable. I'm fixing here just to make the index be consistent with the dynamic allocation index.

Reply:
Ok, I see, thanks!

Checking a bit, I understand this function is used for debugging. Do you know if it's a good state now, giving correct results?

Reply:
I also check this function a bit, it's called in bin/sdl.cc. In principal, if verbose==4, this function should be called. But now someone marked it as // TODO BROKEN // and commented it out for now (and I tested it's indeed broken). I'm afraid if we want to use that, a bit more study and fix need to be done...

Reply:
Ok, then let's not worry about this. I see also that a related issue exists, so that's enough for us to have this in mind for the future.